grafonnet-base/util/grid.libsonnet

```diff
@@ -131,4 +131,69 @@ local d = import 'github.com/jsonnet-libs/docsonnet/doc-util/main.libsonnet';
         panels: root.makePanelGrid(panelsBeforeRowGroups, panelWidth, panelHeight, 0),
       }
     ).panels,
+
+  '#wrapPanels':: d.func.new(
+    |||
+      `wrapPanels` returns an array of `panels` organized in a grid, wrapping up to next 'row' if total width exceeds full grid of 24 columns.
+      'panelHeight' and 'panelWidth' are used unless panels already have height and width defined.
+    |||,
+    args=[
+      d.arg('panels', d.T.array),
+      d.arg('panelWidth', d.T.number),
+      d.arg('panelHeight', d.T.number),
+      d.arg('startY', d.T.number),
+    ],
+  ),
+  wrapPanels(panels, panelWidth=8, panelHeight=8, startY=0):
+    std.foldl(
+      function(acc, panel)
+        local width = if std.objectHas(panel,'gridPos') && std.objectHas(panel.gridPos, 'w') then panel.gridPos.w else panelWidth;
+        local height = if std.objectHas(panel,'gridPos') && std.objectHas(panel.gridPos, 'h') then panel.gridPos.h else panelHeight;
```

Comment:
Instead of std.objectHas you can use std.get() which takes a default value.

Reply:
Thank you, updated. :)

Reply:
Reverted. get() Doesn't work well with nested objects (will fail if gridPos is missing).

Reply:
Here's a small function that gets the Gridpos without if/else:

```jsonnet
{
  getGridPos(panel, defaultWidth, defaultHeight): {
    local gridPos = std.get(panel, 'gridPos', {}),
    w: std.get(gridPos, 'w', defaultWidth),
    h: std.get(gridPos, 'h', defaultHeight),
  },
}
```

Reply:
Updated, but w/o function, as width and height are used in multiple places.